repository/kafka/operator/params.yaml

```diff
@@ -2,52 +2,567 @@ BROKER_COUNT:
     e.g. zk:2181,zk2:2181,zk3:2181
   default: "zk-zk-0.zk-hs:2181,zk-zk-1.zk-hs:2181,zk-zk-2.zk-hs:2181"
   required: true
-KAFKA_ZOOKEEPER_PATH:
+
+ZOOKEEPER_PATH:
   description: "Path inside of KAFKA_ZOOKEEPER_URI to host data"
   default: "/kafka"
-KAFKA_AUTO_CREATE_TOPICS_ENABLE:
+
+AUTO_CREATE_TOPICS_ENABLE:
   default: "true"
-KAFKA_AUTO_LEADER_REBALANCE_ENABLE:
+
+AUTO_LEADER_REBALANCE_ENABLE:
   default: "true"
-KAFKA_BACKGROUND_THREADS:
+
+BACKGROUND_THREADS:
+  description: "The number of threads to use for various background processing tasks"
   default: "10"
-KAFKA_COMPRESSION_TYPE:
+
+COMPRESSION_TYPE:
   default: "producer"
-KAFKA_DELETE_TOPIC_ENABLE:
-  default: "true"
-KAFKA_LEADER_IMBALANCE_CHECK_INTERVAL_SECONDS:
+  description: |
+    Specify the final compression type for a given topic. This configuration accepts the standard compression codecs ('gzip', 'snappy', 'lz4').
+    It additionally accepts 'uncompressed' which is equivalent to no compression; and 'producer' which means retain the original compression codec set by the producer.
+
+DELETE_TOPIC_ENABLE:
+  displayName: "delete.topic.enable"
+  default: "false"
+  description: "Enables delete topic. Delete topic through the admin tool will have no effect if this config is turned off"
+
+DELETE_RECORDS_PURGATORY_PURGE_INTERVAL_REQUESTS:
+  default: "1"
+  description: "The purge interval (in number of requests) of the delete records request purgatory"
+
+LEADER_IMBALANCE_CHECK_INTERVAL_SECONDS:
   default: "300"
-KAKFA_LEADER_IMBALANCE_PER_BROKER_PERCENTAGE:
+  description: "The frequency with which the partition rebalance check is triggered by the controller"
+
+LEADER_IMBALANCE_PER_BROKER_PERCENTAGE:
   default: "10"
-KAFKA_LOG_FLUSH_INTERVAL_MESSAGES:
+  description: "The ratio of leader imbalance allowed per broker. The controller would trigger a leader balance if it goes above this value per broker. The value is specified in percentage."
+
+LOG_FLUSH_INTERVAL_MESSAGES:
+  default: "9223372036854775807"
+  description: "The number of messages accumulated on a log partition before messages are flushed to disk"
+
+LOG_FLUSH_INTERVAL_MS:
+  default: "9223372036854775807"
+  description: "The maximum time in ms that a message in any topic is kept in memory before flushed to disk. If not set, the value in log.flush.scheduler.interval.ms is used"
+
+LOG_FLUSH_OFFSET_CHECKPOINT_INTERVAL_MS:
+  default: "60000"
+  description: "The frequency with which we update the persistent record of the last flush which acts as the log recovery point"
+
+LOG_FLUSH_SCHEDULER_INTERVAL_MS:
+  default: "9223372036854775807"
+  description: "The frequency in ms that the log flusher checks whether any log needs to be flushed to disk"
+
+LOG_FLUSH_START_OFFSET_CHECKPOINT_INTERVAL_MS:
+  default: "60000"
+  description: "The frequency with which we update the persistent record of log start offset"
+
+LOG_RETENTION_BYTES:
+  displayName: "log.retention.bytes"
```

Comment:
if display name is meant as really human readable one I would probably go with "Log retention in bytes" or "Maximum size of the log in bytes" something like that. Or not doing display names at all if you want - they are not used anywhere and right now I am not sure when will they be used. Up to you :)

Reply:
I'm using `displayName` to capture exact same name as used in broker configs https://kafka.apache.org/documentation/#brokerconfigs 

So users can identify how to configure a broker using these params